backend/de.metas.business/src/main/java/de/metas/order/location/adapter/OrderLineMainLocationAdapter.java

```diff
@@ -107,4 +117,21 @@ public void setFromOrderHeader(@NonNull final I_C_Order order)
 
 		setFrom(orderLocation);
 	}
+
+	public void setLocationAndResetRenderedAddress(@NonNull final BPartnerLocationAndCaptureId from)
+	{
+		setC_BPartner_Location_ID(from != null ? from.getBPartnerLocationRepoId() : -1);
+		setC_BPartner_Location_Value_ID(from != null ? from.getLocationCaptureRepoId() : -1);
+		setBPartnerAddress(null);
+	}
+
+	public void setBPartnerAndLocation(@NonNull final BPartnerId from)
```

Comment:
i think this adapter class shouldn't look up `I_C_BPartner_Location`s. please call it with a `BPartnerLocationId` that you computed elsewhere